solenspipe/solenspipe.py

```diff
@@ -194,28 +298,68 @@ def get_qfunc(px,ucls,mlmax,est1,Al1=None,est2=None,Al2=None,R12=None):
     else:
         bh = False
 
-    assert est1 in ['TT','TE','EE','EB','TB','MV','MVPOL'] # TODO: add other
-    qfunc1 = lambda X,Y: qe.qe_all(px,ucls,mlmax,
-                                   fTalm=Y[0],fEalm=Y[1],fBalm=Y[2],
-                                   estimators=[est1],
-                                   xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[est1]
+    assert est1 in ['TT','TE','EE','EB','TB','MV','MVPOL','SHEAR'] # TODO: add other
+    if est1=='SHEAR':
+        qfunc1 = lambda X,Y: qe.qe_shear(px,mlmax,
+                            Talm=X[0],fTalm=Y[1])
+    else:
+        qfunc1 = lambda X,Y: qe.qe_all(px,ucls,mlmax,
+                                    fTalm=Y[0],fEalm=Y[1],fBalm=Y[2],
+                                    estimators=[est1],
+                                    xfTalm=X[0],xfEalm=X[1],xfBalm=X[2])[est1]
 
     if bh:
-        assert est2 in ['SRC'] # TODO: add mask
-        qfunc2 = lambda X,Y: qe.qe_pointsources(px,mlmax,fTalm=Y[0],xfTalm=X[0])
+        assert est2 in ['SRC','PH','MASK'] # TODO: add mask
+        if est2=='SRC' and profile is None:
```

Comment:
I don't think we need to separate the point-source and profile-hardened cases here? I think `qfunc2 = lambda X,Y: qe.qe_source(px,mlmax,Y[0],profile=profile,xfTalm=X[0])` works for both since if `profile=None` we'll get the point-source hardened estimator?